jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletApiResponse.java

```diff
@@ -0,0 +1,649 @@
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.channels.IllegalSelectorException;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+import jakarta.servlet.ServletOutputStream;
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpSession;
+import org.eclipse.jetty.ee10.servlet.writer.EncodingHttpWriter;
+import org.eclipse.jetty.ee10.servlet.writer.Iso88591HttpWriter;
+import org.eclipse.jetty.ee10.servlet.writer.ResponseWriter;
+import org.eclipse.jetty.ee10.servlet.writer.Utf8HttpWriter;
+import org.eclipse.jetty.http.HttpCookie;
+import org.eclipse.jetty.http.HttpFields;
+import org.eclipse.jetty.http.HttpGenerator;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpScheme;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.http.HttpURI;
+import org.eclipse.jetty.http.HttpVersion;
+import org.eclipse.jetty.http.MimeTypes;
+import org.eclipse.jetty.io.RuntimeIOException;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.session.Session;
+import org.eclipse.jetty.session.SessionManager;
+import org.eclipse.jetty.util.Blocker;
+import org.eclipse.jetty.util.FutureCallback;
+import org.eclipse.jetty.util.StringUtil;
+import org.eclipse.jetty.util.URIUtil;
+
+/**
+ * The Jetty low level implementation of the ee10 {@link HttpServletResponse} object.
+ *
+ * <p>
+ *     This provides the bridges from Servlet {@link HttpServletResponse} to the Jetty Core {@link Response} concepts (provided by the {@link ServletContextResponse})
+ * </p>
+ */
+public class ServletApiResponse implements HttpServletResponse
+{
+    private static final int MIN_BUFFER_SIZE = 1;
+    private static final EnumSet<ServletContextResponse.EncodingFrom> LOCALE_OVERRIDE = EnumSet.of(
+        ServletContextResponse.EncodingFrom.NOT_SET,
+        ServletContextResponse.EncodingFrom.DEFAULT,
+        ServletContextResponse.EncodingFrom.INFERRED,
+        ServletContextResponse.EncodingFrom.SET_LOCALE
+    );
+
+    private final ServletContextResponse _response;
+
+    protected ServletApiResponse(ServletContextResponse response)
+    {
+        _response = response;
+    }
+
+    public ServletContextResponse getResponse()
+    {
+        return _response;
+    }
+
+    @Override
+    public void addCookie(Cookie cookie)
+    {
+        if (StringUtil.isBlank(cookie.getName()))
+            throw new IllegalArgumentException("Cookie.name cannot be blank/null");
+
+        // new style cookie, everything is an attribute
+        addCookie(new HttpCookie(
+            cookie.getName(),
+            cookie.getValue(),
+            cookie.getVersion(),
+            cookie.getAttributes()));
```

Comment:
I'm a little cautious about this, as if there are attributes, then we will always create a new map.
But I think most usages will have no attributes.... so maybe OK. But maps are expensive... TreeMaps more so.

Reply:
Actually most usages will have attributes, since path & domain are implemented as attributes.

Reply:
We can address this concern in the followup HttpCookie cleanup PR.
I have a few ideas on how to minimize this concern when going from Servlet Cookie API to our HttpCookie API.